### DISCUSSION TITLE
Steam(Deck) shortcuts and controller

### DISCUSSION THREAD

**Discussion by dpetroff** [lutris/lutris#5328] [Q&A]
Is the logic behind https://github.com/lutris/lutris/blob/351be70065a35c8c1d3d13b9b133d6e2631d8157/lutris/util/steam/shortcut.py#L115-L130 documented somewhere?

I'm not opening this as an issue because I gave up and installed Windows 11 on my Steam Deck, so I can't test it anymore, but I'm fairly sure since the executable is essentially always `flatpak` or `lutris`, Steam can never actually know which game I run (and consequently which controller profile to activate), unless, for some reason, I launch it from Big Picture mode. If I remember right, launching from Steam was not enough, it had to be Big Picture specifically.

Maybe someone else with a Steam Deck can confirm the details but I'm also curious to understand how (we think) Steam detects non-Steam games to be running.

**Comment by strycore** [comment]
Steam Input works for each individual Lutris game. The executable name doesn't matter.

> Windows 11 on my Steam Deck

Good luck with that...

**Reply by dpetroff** [reply]
> Good luck with that...

It's working surprisingly well so far. Even the battery life is not noticeably worse. You lose some of the Steam sliders for frame limiters and that but it more than makes up for it by not having to fiddle with Wine.

Don't get me wrong though, I want gaming on Linux to get better. I have to admit I've not tried it on something other than Steam Deck, but my guess is Desktop Mode is representative of how other distros work, and I can confirm that the controller scheme does not switch from Desktop to the specific game's controller scheme unless I launch the game via Steam, whether that's from the UI (BigPicture or not) or via a desktop shortcut specifically launching the non-steam app via Steam, which is a bit fiddly to create.

Considering you have a mouse and keyboard to control other distros rather than a steam controller with Desktop-specific controller scheme, I guess this is really a Steam-Deck-specific problem. I would suggest that on Steam Deck (or alternatively when the option to create a steam shortcut is selected), desktop shortcuts should be created on top of the Steam shortcut to launch the game via Steam, otherwise it definitely does not switch the controller scheme from Desktop to the game's scheme.

Sorry if this is already the case, I just wanted to make sure this doesn't get forgotten about as it would be a huge usability improvement.
